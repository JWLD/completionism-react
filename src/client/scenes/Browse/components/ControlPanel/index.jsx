import React from 'react'

const ControlPanel = () => <h1>I am a control panel yay!</h1>

export default ControlPanel
