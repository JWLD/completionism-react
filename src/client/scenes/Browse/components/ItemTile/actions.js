import * as ACTIONS from 'constants/action_types'

export const setActiveItem = itemId => ({
  type: ACTIONS.SET_ACTIVE_ITEM,
  payload: itemId
})
