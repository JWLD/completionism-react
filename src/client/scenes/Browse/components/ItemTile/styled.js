import styled from 'styled-components'
import FaCheckCircle from 'react-icons/lib/fa/check-circle'
import FaTimesCircle from 'react-icons/lib/fa/times-circle'

import { mixins, vars } from 'style'

const itemTilePadding = 0.5
const iconSize = vars.sizing.itemHeight - itemTilePadding * 2

export const ItemTile = styled.li`
  ${mixins.itemListItem};

  justify-content: space-between;
  height: ${vars.sizing.itemHeight}rem;
  padding: ${itemTilePadding}rem;
  background-color: #444;

  :not(:last-child) {
    margin-bottom: 1rem;
  }

  :hover {
    cursor: pointer;
    background-color: #555;
  }
`

export const ItemIcon = styled.i`
  height: ${iconSize}rem;
  width: ${iconSize}rem;
  margin-right: ${itemTilePadding * 2}rem;

  border-radius: 0.4rem;
  background-size: cover;
  background-color: #222;
`

export const ItemTitle = styled.span`
  flex-grow: 1;
  color: ${props => vars.colours[`q${props.quality}`]};
  font: 2.25rem DinRC;
  text-align: center;
`

export const ProgressIconWrap = styled.div`
  height: ${iconSize}rem;
  width: ${iconSize}rem;
  padding: 0.5rem;
  border-radius: 0.4rem;
  background-color: #333;
`

export const CheckIcon = ProgressIconWrap.withComponent(FaCheckCircle).extend`
  color: ${vars.colours.pos};
`

export const CrossIcon = ProgressIconWrap.withComponent(FaTimesCircle).extend`
  color: ${vars.colours.neg};
`
