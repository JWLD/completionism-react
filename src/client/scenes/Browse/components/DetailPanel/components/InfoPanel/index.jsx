import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import { NPC_RENDER_URLS } from 'constants/urls'
import { itemDataSelector } from 'InfoPanel/selectors'

import * as SC from 'InfoPanel/styled'

const InfoPanel = ({ itemData }) => {
  const imageUrl =
    itemData.displayId && `${NPC_RENDER_URLS.zoom}${itemData.displayId}.jpg`

  return (
    <SC.InfoPanel>
      <SC.ItemName>{itemData.name || 'Select An Item'}</SC.ItemName>
      <SC.ItemImage imageUrl={imageUrl} />
      <SC.WowheadLink
        href={`http://www.wowhead.com/spell=${itemData.id}`}
        target="_blank">
        View on Wowhead
      </SC.WowheadLink>
    </SC.InfoPanel>
  )
}

InfoPanel.propTypes = {
  itemData: PropTypes.object.isRequired
}

const mapStateToProps = (state, props) => ({
  itemData: itemDataSelector(state, props)
})

const ReduxComponent = connect(mapStateToProps)(InfoPanel)

export default withRouter(ReduxComponent)
