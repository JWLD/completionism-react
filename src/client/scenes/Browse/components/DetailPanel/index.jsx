import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import { controlPanelSelector, itemDataSelector } from 'DetailPanel/selectors'
import { NPC_RENDER_URLS } from 'constants/urls'
import { toggleControlPanel } from 'DetailPanel/actions'

import * as SC from 'DetailPanel/styled'

export const DetailPanel = props => {
  const imageUrl = props.itemData.displayId
    ? `${NPC_RENDER_URLS.zoom}${props.itemData.displayId}.jpg`
    : null

  const title = props.controlPanelIsActive
    ? 'Control Panel'
    : props.itemData.name || 'Select An Item'

  const content = props.controlPanelIsActive ? null : (
    <SC.ItemImage imageUrl={imageUrl} />
  )

  return (
    <SC.DetailPanel>
      <SC.TopBar>
        <SC.Title>{title}</SC.Title>
        <SC.ToggleWrap onClick={() => props.toggleControlPanel()}>
          <SC.CogIcon active={props.controlPanelIsActive} />
          <SC.InfoIcon active={!props.controlPanelIsActive} />
        </SC.ToggleWrap>
      </SC.TopBar>
      {content}
    </SC.DetailPanel>
  )
}

DetailPanel.propTypes = {
  controlPanelIsActive: PropTypes.bool.isRequired,
  itemData: PropTypes.object.isRequired,
  toggleControlPanel: PropTypes.func.isRequired
}

const mapStateToProps = (state, props) => ({
  controlPanelIsActive: controlPanelSelector(state),
  itemData: itemDataSelector(state, props)
})

const mapDispatchToProps = {
  toggleControlPanel
}

const reduxComponent = connect(mapStateToProps, mapDispatchToProps)(DetailPanel)

export default withRouter(reduxComponent)
