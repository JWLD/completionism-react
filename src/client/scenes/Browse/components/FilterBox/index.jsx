import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { resetFilter, setFilter } from 'FilterBox/actions'

import * as SC from 'FilterBox/styled'

export const FilterBox = props => (
  <SC.FilterWrap>
    <SC.FilterInput
      onChange={e => props.setFilter(e.target.value)}
      value={props.filterValue}
      placeholder="Filter"
    />
    <SC.ClearInputBtn
      active={props.filterValue}
      onClick={() => props.resetFilter()}
    />
  </SC.FilterWrap>
)

FilterBox.propTypes = {
  filterValue: PropTypes.string.isRequired,
  resetFilter: PropTypes.func.isRequired,
  setFilter: PropTypes.func.isRequired
}

export const mapStateToProps = state => ({
  filterValue: state.browse.filter
})

const mapDispatchToProps = {
  resetFilter,
  setFilter
}

export default connect(mapStateToProps, mapDispatchToProps)(FilterBox)
