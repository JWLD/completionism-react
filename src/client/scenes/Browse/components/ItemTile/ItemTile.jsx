import React from 'react'
import PropTypes from 'prop-types'

import { ICON_URLS } from 'constants/urls'
import * as SC from './styled'

const isCollected = props => {
  const { category, id, storageData } = props

  let collected = false

  if (category === 'pets') {
    collected = storageData.find(el => el.id === id)
  } else {
    collected = storageData.includes(id)
  }

  return collected
}

const ItemTile = props => {
  const { icon, name, quality } = props

  const itemIcon = icon
    ? { backgroundImage: `url(${ICON_URLS.large}${icon}.jpg)` }
    : null

  return (
    <SC.ItemTile>
      <SC.ItemIcon style={itemIcon} />
      <SC.ItemTitle quality={quality}>{name}</SC.ItemTitle>
      {isCollected(props) ? <SC.CheckIcon /> : <SC.CrossIcon />}
    </SC.ItemTile>
  )
}

ItemTile.defaultProps = {
  icon: null
}

ItemTile.propTypes = {
  icon: PropTypes.string,
  name: PropTypes.string.isRequired,
  quality: PropTypes.number.isRequired
}

export default ItemTile
