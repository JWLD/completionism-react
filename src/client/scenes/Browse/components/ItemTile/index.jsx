import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { ICON_URLS } from 'constants/urls'
import { setActiveItem } from 'ItemTile/actions'

import * as SC from 'ItemTile/styled'

export const ItemTile = props => {
  const { collected, icon, id, name, quality } = props

  const iconUrl = icon ? `url(${ICON_URLS.large}${icon}.jpg)` : 'none'
  const progressIcon = collected ? <SC.CheckIcon /> : <SC.CrossIcon />

  return (
    <SC.ItemTile onClick={() => props.setActiveItem(id)}>
      <SC.ItemIcon iconUrl={iconUrl} />
      <SC.ItemTitle quality={quality}>{name}</SC.ItemTitle>
      {progressIcon}
    </SC.ItemTile>
  )
}

ItemTile.defaultProps = {
  icon: null
}

ItemTile.propTypes = {
  collected: PropTypes.bool.isRequired,
  icon: PropTypes.string,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  quality: PropTypes.number.isRequired,
  setActiveItem: PropTypes.func.isRequired
}

const mapDispatchToProps = {
  setActiveItem
}

export default connect(null, mapDispatchToProps)(ItemTile)
