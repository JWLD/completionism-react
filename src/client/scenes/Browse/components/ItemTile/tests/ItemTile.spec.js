import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { ItemTile } from 'scenes/Browse/components/ItemTile'

Enzyme.configure({ adapter: new Adapter() })

let mockProps = {}

beforeEach(() => {
  mockProps = {
    collected: true,
    name: 'foo',
    quality: 1,
    setActiveItem: jest.fn()
  }
})

describe('<ItemTile />', () => {
  it('renders correctly', () => {
    const component = shallow(<ItemTile {...mockProps} />)

    expect(component).toMatchSnapshot()
  })

  it('renders when [collected] prop is false', () => {
    mockProps.collected = false
    const component = shallow(<ItemTile {...mockProps} />)

    expect(component).toMatchSnapshot()
  })

  it('renders when passed an [icon] prop', () => {
    mockProps.icon = 'foo'
    const component = shallow(<ItemTile {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
