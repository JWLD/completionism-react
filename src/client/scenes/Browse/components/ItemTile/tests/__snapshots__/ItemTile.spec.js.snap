// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemTile /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTile
    collected={true}
    icon={null}
    name="foo"
    quality={1}
    setActiveItem={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.i
          iconUrl="none"
        />,
        <ItemTitle
          quality={1}
        >
          foo
        </ItemTitle>,
        <Styled(FaCheckCircle) />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconUrl": "none",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "foo",
          "quality": 1,
        },
        "ref": null,
        "rendered": "foo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.i
            iconUrl="none"
          />,
          <ItemTitle
            quality={1}
          >
            foo
          </ItemTitle>,
          <Styled(FaCheckCircle) />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "iconUrl": "none",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo",
            "quality": 1,
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ItemTile /> renders when [collected] prop is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTile
    collected={false}
    icon={null}
    name="foo"
    quality={1}
    setActiveItem={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.i
          iconUrl="none"
        />,
        <ItemTitle
          quality={1}
        >
          foo
        </ItemTitle>,
        <Styled(FaTimesCircle) />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconUrl": "none",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "foo",
          "quality": 1,
        },
        "ref": null,
        "rendered": "foo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.i
            iconUrl="none"
          />,
          <ItemTitle
            quality={1}
          >
            foo
          </ItemTitle>,
          <Styled(FaTimesCircle) />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "iconUrl": "none",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo",
            "quality": 1,
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ItemTile /> renders when passed an [icon] prop 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTile
    collected={true}
    icon="foo"
    name="foo"
    quality={1}
    setActiveItem={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.i
          iconUrl="url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)"
        />,
        <ItemTitle
          quality={1}
        >
          foo
        </ItemTitle>,
        <Styled(FaCheckCircle) />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconUrl": "url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "foo",
          "quality": 1,
        },
        "ref": null,
        "rendered": "foo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.i
            iconUrl="url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)"
          />,
          <ItemTitle
            quality={1}
          >
            foo
          </ItemTitle>,
          <Styled(FaCheckCircle) />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "iconUrl": "url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo",
            "quality": 1,
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
